src/main/java/ch/unisg/ics/interactions/wot/td/clients/UriTemplate.java

```diff
@@ -14,6 +14,8 @@ public UriTemplate(String expression) {
   }
 
   static List<String> extract(String path) {
+    path = path.replaceAll("%7B", "{");
+    path = path.replaceAll("%7D", "}");
```

Comment:
while this can work, it will not catch other issues with url encoded paths.

I'd go for something that uses the URLDecoder:

```suggestion
    path = URLDecoder.decode(path, "UTF-8")
```